neptune-core/src/application/rpc/server.rs

```diff
@@ -345,6 +346,11 @@ pub trait RPC {
     /// ```
     async fn confirmations(token: auth::Token) -> RpcResult<Option<BlockHeight>>;
 
+    /// Return the most recently generated generation address. Does not add a
+    /// new key to the wallet, and does not change the derivation index.
+    async fn latest_generation_address(token: auth::Token)
```

Comment:
This is not a forward thinking method as it does not support any future key types.   I would suggest instead:

```
fn latest_address(key_type: KeyType)
```